superset-frontend/src/components/AnchorLink/index.tsx

```diff
@@ -69,29 +53,35 @@ class AnchorLink extends React.PureComponent {
     }
   }
 
-  render() {
-    const { anchorLinkId, filters, showShortLinkButton, placement } =
-      this.props;
-    return (
-      <span className="anchor-link-container" id={anchorLinkId}>
-        {showShortLinkButton && (
-          <URLShortLinkButton
-            url={getDashboardUrl({
-              pathname: window.location.pathname,
-              filters,
-              hash: anchorLinkId,
-            })}
-            emailSubject={t('Superset chart')}
-            emailContent={t('Check out this chart in dashboard:')}
-            placement={placement}
-          />
-        )}
-      </span>
-    );
-  }
-}
+  useEffect(() => {
+    if (inFocus) {
+      scrollToView();
+    }
+  }, [inFocus])
```

Comment:
Resources referred to replace the `UNSAFE_componentWillReceiveProps`:

- https://codezup.com/unsafe-componentwillreceiveprops-in-react-hook-functional/
- https://reactjs.org/docs/react-component.html